### ISSUE TITLE
Add reset for Gen 2 roamer encounter

### ISSUE THREAD

**Issue by brainard52** [kwsch/PKHeX#3162]
Later gens have an option to edit or reset the encounter for roamers. Since it's now possible to use Lua scripts to reset for shinies before loading the save into the Virtual Console for transfer to modern games, I think this would be a great addition to gen 2.

**Comment by kwsch** [comment]
Research required:

* Link to the Lua scripts
* Locate the corresponding offsets in the save file (compared to the Lua script changing RAM) for all languages

**Comment by brainard52** [comment]
Lua scripts: https://github.com/wwwwwwzx/gsclua
The ROM can be extracted from VC by means of GodMode9. The save can be exctracted from VC by means of CheckPoint. The save can be injected again by converting with https://sav2vc.fm1337.com/ and then replacing the sav.dat in the appropriate checkpoint backup folder on the SDCard and restoring the save backup.

**Comment by kwsch** [comment]
Pretty sure this was implemented via the above commit; please let me know if not.

**Comment by brainard52** [comment]
<img width="522" alt="Screen Shot 2021-04-04 at 7 53 10 PM" src="https://user-images.githubusercontent.com/1023838/113532224-b382a180-957f-11eb-9173-6140e6de8de4.png">

I unchecked "Legendary Beasts Disappeared from Burned Tower" and exported the save. This is what I got.

**Comment by kwsch** [comment]
@brainard52 Did you reset the event outside of the map, and then entered the area later?

> "Object event." | オブジェイベント | Overworld | This text will appear when trying to talk to an object which has no event set for it.

**Comment by brainard52** [comment]
> @brainard52 Did you reset the event outside of the map, and then entered the area later?
> 
> > "Object event." | オブジェイベント | Overworld | This text will appear when trying to talk to an object which has no event set for it.

I was in front of the Pokemon Center in the same town. Does that count as outside the map?

**Comment by kwsch** [comment]
Added the recent resources, so they now show up in the second tab.
https://github.com/kwsch/PKHeX/commit/d0258fa2f7f9637c23773b540b6683039d3da52f

Changing those /should/ work in making the event reset.
(dev builds are available at pporg once the commit is built)

**Comment by brainard52** [comment]
https://user-images.githubusercontent.com/1023838/113548592-a75d0b00-95a4-11eb-9586-039469bcdba7.mov
(might not play nicely with people that have photosensitivity)

The newest change allowed me to fall in the hole and set the beasties loose again. However, I cannot encounter them again for some reason. Is there some flag which says whether I've encountered a beast in the wild? I believe it stores their stats or something somehow, since they're always the same if you save after encountering one.

**Comment by Lusamine** [comment]
I tested this on a Crystal save that had already captured all 3 legendary beasts. The only two changes I made were: I reset the "Legendary Beasts Disappeared from Burned Tower" flag and I set the "Legendary Beasts (Burned Tower)" constant to "Not Awakened".  This appropriately triggered the cutscene where the beasts run away in Burned Tower and Eusine appears.  Afterwards, Raikou appeared on the Pokedex map, and I was able to encounter Raikou manually on a route.  Maybe you haven't run the LUA script long enough?
